### ISSUE TITLE
refactor Block to derive `include_in_legacy` from `legacy_id`

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#14262]
follow-up from #14222 - really, we don't want to be holding `include_in_legacy` property on `Block`, when instead we could be inspecting `kinto_id`.  If it's set it means there's a matching record in the legacy blocklist so it's included (`include_in_legacy is True`); if it's not set there isn't a matching record in the legacy blocklist (`include_in_legacy is False`).

It's easy enough to drop the model field and replace it with a `@property`, but when the `blocklist_legacy_submit` waffle switch is enabled we also want to be able to change it.  This means the `BlocklistSubmission` object still needs to store the property value.  Added complication is we cheat with the change page for a `Block` and submit the form directly into `BlocklistSubmission.add_view`, which relies on the Block change form having the same fields.

**Comment by AlexandraMoga** [comment]
@eviljeff from what I understand, I should be checking that the blocks added in kinto are still being identified as legacy blocks but I don't quite get how the `include_in_legacy` waffle switch is supposed to behave now.

**Comment by eviljeff** [comment]
@AlexandraMoga yes, auto-imported blocks from the legacy blocklist should continue to be identified with the `include_in_legacy` checkbox being enabled.  And when the `blocklist_legacy_submit` waffle switch is enabled the checkbox should still work - i.e. you can select it to have the block submitted to the v2 blocklist, and deselect it to have it removed.

Internally, the difference is the checkbox should be a reflection that the `legacy_id` is set on the Block (which is set either when it's imported, or when a block is submitted to the legacy blocklist and we get the id back from their api), but from the admin, it should look and work the same*

*there may be some changes with older blocks that were created/updated before `blocklist_legacy_submit` being disabled stopped the checkbox from being changed.  That's fine - they were wrong.

**Comment by AlexandraMoga** [comment]
Verified fixed on stage:

- with `blocklist_legacy_submit` disabled:
  - auto-imported blocks will have a Legacy id to mark them as imports from kinto (this replaces the selected checkbox that we used to show previously)
  - auto-imported blocks can't be edited through the blocklist admin

- with `blocklist_legacy_submit` enabled:
  - auto-imported blocks can be edited through the blocklist admin
  - the legacy checbox can be enabled/disabled

So, to sum up, except for the legacy checkbox being replaced with the Legacy ID for imported blocks, I did not see any other changes in behavior.

**Comment by eviljeff** [comment]
> except for the legacy checkbox being replaced with the Legacy ID for imported blocks

Can you show me what you mean?  This wasn't intentional (though it doesn't sound too bad)

**Comment by AlexandraMoga** [comment]
@eviljeff here it is:

![image](https://user-images.githubusercontent.com/31961530/88922050-3a0c1400-d278-11ea-811b-c5e91935c157.png)

This is for a non-editable imported block.
